### PR TITLE
[IOTDB-4039] Unify thread-name of ClientPools

### PR DESCRIPTION
**Pull request by RYH61**
1、We can view online threads：
Example：

图片
图片
图片

meanwhile
图片

2、Delete a duplicate class
AsyncDataNodeInternalServiceClientPoolFactory

### CODE REVIEW

**Review by CRZbulabula — APPROVED** [review]
LGTM

File: node-commons/src/main/java/org/apache/iotdb/commons/client/AsyncBaseClientFactory.java [hunk 1]
```diff
@@ -43,5 +47,11 @@ protected AsyncBaseClientFactory(
         logger.error("Cannot create Async client factory", e);
       }
     }
+    synchronized (this) {
+      Thread.getAllStackTraces().keySet().stream()
+          .filter(thread -> thread.getName().contains(THRIFT_THREAD_NAME))
+          .collect(Collectors.toList())
+          .forEach(thread -> thread.setName(threadName + "-" + thread.getId()));
```
Comment (qiaojialin) [inline node-commons/src/main/java/org/apache/iotdb/commons/client/AsyncBaseClientFactory.java:54]:
```suggestion
          .forEach(thread -> thread.setName(threadName  + "-selector"  + "-" + thread.getId()));
```

Comment (RYH61) [reply node-commons/src/main/java/org/apache/iotdb/commons/client/AsyncBaseClientFactory.java]: fixed~

File: node-commons/src/main/java/org/apache/iotdb/commons/client/AsyncBaseClientFactory.java [hunk 2]
```diff
@@ -25,15 +25,19 @@
...
 import java.io.IOException;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
 
 public abstract class AsyncBaseClientFactory<K, V> extends BaseClientFactory<K, V> {
 
   private static final Logger logger = LoggerFactory.getLogger(AsyncBaseClientFactory.class);
   protected TAsyncClientManager[] tManagers;
   protected AtomicInteger clientCnt = new AtomicInteger();
+  private static final String THRIFT_THREAD_NAME = "TAsyncClientManager#SelectorThread";
```
Comment (qiaojialin) [inline node-commons/src/main/java/org/apache/iotdb/commons/client/AsyncBaseClientFactory.java:35]: This is defined by ourselves, we should have the ability to name this selector thread at init.
Instead traverse all thread to modify a name that we give to it.

**Review by qiaojialin — APPROVED** [review]

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**